std/typecons.d

```diff
@@ -3109,6 +3109,71 @@ struct Nullable(T)
     {
         return isNull ? fallback : _value.payload;
     }
+
+    // Range interface
+    /**
+     * Returns whether or not any values remain in this range.
+     */
+    alias empty = isNull;
+    /**
+     * Advances this range.
+     */
+    alias popFront = nullify;
+    /// ditto
+    alias popBack = nullify;
+    /**
+     * Returns the range's current value.
+     */
+    @property ref inout(T) front() inout @safe pure nothrow
+    {
+        return get();
+    }
+    /// ditto
+    alias back = front;
+    ///
+    @property inout(typeof(this)) save()() inout
+    {
+        return this;
+    }
+    ///
+    inout(typeof(this)) opSlice()() inout
+    {
+        return this;
+    }
+    ///
+    inout(typeof(this)) opSlice()(size_t from, size_t to) inout
+    {
+        import core.exception : RangeError;
+        if(from > length || to > length || to < from) {
+            throw new RangeError(__FILE__, __LINE__);
+        }
```

Comment:
This could be more succinctly expressed as an `in` contract:

```d
in (from <= length && to <= length && from <= to)
```

Reply:
True, but I don't believe we can throw anything other than an AssertError from there.

Reply:
The result of a contract failure depends on the `-checkaction` and `-betterC` switches. By default, it throws an `AssertError`, but that's not the only possible behavior.

Reply:
But shouldn't it throw a RangeError to be consistent with other ranges?

Reply:
Huh... I could've sworn they used RangeError. I remember reviewing them last night, but I guess I must've been too tired.